packets1/auth_test.go

```diff
@@ -1,13 +1,117 @@
 package packets1
 
 import (
+	"bytes"
+	"fmt"
+	"reflect"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+
+	pkts "github.com/energomonitor/bisquitt/packets"
 )
 
-func TestAuthMarshal(t *testing.T) {
+func TestAuthPlainConstructor(t *testing.T) {
+	assert := assert.New(t)
+
+	user := "test-user"
+	password := []byte("test-password")
+	pkt := NewAuthPlain(user, password)
+
+	if pkt == nil {
+		t.Fatal("New packet should not be nil")
+	}
+
+	assert.Equal("*packets1.Auth", reflect.TypeOf(pkt).String(), "Type should be Connect")
+	assert.Equal(uint8(0), pkt.Reason, "Bad Reason value")
+	assert.Equal("PLAIN", pkt.Method, "Bad Method value")
+	assert.Equal(
+		[]byte(fmt.Sprintf("\x00%s\x00%s", user, password)),
+		pkt.Data, "Bad Data value")
+}
+
+func TestAuthPlainMarshal(t *testing.T) {
 	pkt1 := NewAuthPlain("test-user", []byte("test-password"))
 	pkt2 := testPacketMarshal(t, pkt1)
 	assert.Equal(t, pkt1, pkt2.(*Auth))
 }
+
+func TestAuthUnmarshalInvalid(t *testing.T) {
+	assert := assert.New(t)
+
+	// Packet too short - Method Length missing.
+	buff := bytes.NewBuffer([]byte{
+		3,               // Length
+		byte(pkts.AUTH), // MsgType
+		0,               // Reason
+		// Method Length missing
+	})
+	_, err := ReadPacket(buff)
+	if assert.Error(err) {
+		assert.Contains(err.Error(), "bad AUTH packet length")
+	}
+
+	// Packet too short - Method missing.
+	buff = bytes.NewBuffer([]byte{
+		4,               // Length
+		byte(pkts.AUTH), // MsgType
+		0,               // Reason
+		1,               // Method Length
+		// Method missing
+	})
+	_, err = ReadPacket(buff)
+	if assert.Error(err) {
+		assert.Contains(err.Error(), "bad AUTH packet length")
+	}
+}
+
+func TestAuthPlain(t *testing.T) {
```

Comment:
Wouldn’t this be better named `TestAuthDecodePlain`?

Reply:
Probably yes, done. I have also found a typo "Type should be Connect". I'm fixing it in the same force push.